### PR TITLE
Email - Blocklist

### PR DESCRIPTION
**Pull request by lukmzig**
## Changes in this pull request
Part of https://github.com/pimcore/studio-ui-bundle/issues/305

## Additional info
- endpoint for listing blocklist entries
- endpoint for deleting entry
- endpoint for adding a blocklist entry

### CODE REVIEW

File: src/Email/Controller/Blocklist/DeleteController.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    #[SuccessResponse(
+        description: 'Successfully deleted blocklist entry',
+    )]
+    #[DefaultResponses([
+        HttpResponseCodes::UNAUTHORIZED,
+        HttpResponseCodes::NOT_FOUND,
+    ])]
+    public function deleteBlocklistEntry(#[MapQueryParameter] string $email): Response
+    {
+        $this->repository->deleteEntry($email);
```
Comment (mattamon) [inline src/Email/Controller/Blocklist/DeleteController.php]: No service layer? For consistency and to reinforce the blueprint I think we should move it to a service layer like the rest :)

File: src/Email/Controller/Blocklist/DeleteController.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * @internal
+ */
+final class DeleteController extends AbstractApiController
+{
+    public function __construct(
+        SerializerInterface $serializer,
+        private readonly BlocklistRepositoryInterface $repository
```
Comment (mattamon) [inline src/Email/Controller/Blocklist/DeleteController.php]: Naming here in the collection it is called `blocklistRepository` and here only `repository`

File: src/OpenApi/Config/Tags.php [hunk 3]
```diff
@@ -97,6 +101,7 @@ enum Tags: string
     case DataObjects = 'DataObjects';
     case Dependencies = 'Dependencies';
     case Elements = 'Elements';
+    case Emails = 'Emails';
```
Comment (mattamon) [inline src/OpenApi/Config/Tags.php]:
```suggestion
    case Emails = 'E-Mails';
```

File: src/Email/Repository/BlocklistRepository.php [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        $list = [];
+        $listing = $this->getListing($parameters, $email);
+        foreach ($listing as $listEntry) {
+            $entry = new BlocklistEntry(
+                $listEntry->getAddress(),
+                $listEntry->getCreationDate(),
+                $listEntry->getModificationDate()
+            );
+
+            $this->eventDispatcher->dispatch(
```
Comment (mattamon) [inline src/Email/Repository/BlocklistRepository.php]: Same here, according to the blueprint we should dispatch events only in our service layer :)

**Review by mattamon — CHANGES_REQUESTED** [review]
Rest LGTM! (s)

**Comment by lukmzig** [comment]
@mattamon good point, thank you :) I added the service layer to adhere to the blueprint

**Pull request merged**